pkg/store/snapshot/device/store.go

```diff
@@ -316,7 +316,7 @@ func (s *atomixStore) Load(deviceID device.VersionedID) (*devicesnapshot.Snapsho
 	defer cancel()
 
 	entry, err := s.snapshots.Get(ctx, string(deviceID))
-	if err != nil {
+	if err != nil && !errors.IsNotFound(err) {
```

Comment:
This `errors.IsNotFound(err)` condition will _only_ work using the Atomix `errors` package since it's an Atomix error that's returned by `snaphsots.Get`. If you convert it into an ONOS error first, then you can use the ONOS `errors` package to check `IsNotFound`.